litmus/litmus.c

```diff
@@ -314,6 +316,50 @@ asmlinkage long sys_null_call(cycles_t __user *ts)
 	return ret;
 }
 
+/*	This is a helper syscall to query if the budget has been exhausted.
+ *
+ *	returns 0 on successful query
+ *	returns EPERM if task is not a real-time task.
+ *      returns EFAULT if &job is not a valid pointer.
+ */
+asmlinkage long sys_query_budget_exhausted(int __user *flag)
+{
+	int retval = -EPERM;
+	if (is_realtime(current))
+		retval = put_user(budget_exhausted(current), flag);
```

Comment:
`budget_exhausted()` relies on the scheduler's time-tracking services. In particular, in order for `budget_exhausted()` to be correct, the scheduler must have "recently" charged execution costs. This, however, is done only prior to a call to `schedule()`, unless it is explicitly triggered. I'm afraid this patch works superficially, but reports wrong values if you have jobs that run without preemption for a couple of milliseconds.